crates/bevy_window/src/window.rs

```diff
@@ -1,17 +1,13 @@
 use bevy_math::{IVec2, Vec2};
-use bevy_utils::{tracing::warn, Uuid};
+use bevy_utils::{tracing::warn, Id, IdType};
 use raw_window_handle::RawWindowHandle;
 
-#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
-pub struct WindowId(Uuid);
+#[derive(Id, Copy, Clone, Hash, Eq, PartialEq, Debug)]
+pub struct WindowId(pub IdType);
 
 impl WindowId {
-    pub fn new() -> Self {
-        WindowId(Uuid::new_v4())
-    }
-
     pub fn primary() -> Self {
-        WindowId(Uuid::from_u128(0))
+        WindowId(0)
     }
```

Comment:
This change is problematic, because the first WindowId generated via new() will be the primary window.
Should we really encode this information into the id itself anyway?

Reply:
Maybe use `!0` as primary window id?